pkg/controller/shared_state.go

```diff
@@ -1227,16 +1233,17 @@ func (c *SharedState) getNodesForReplica(ctx context.Context, volumeName string,
 	}
 
 	filtered := []string{}
-	numFiltered := 0
 	for _, node := range nodes {
-		if numFiltered >= numReplica {
-			break
-		}
```

Comment:
Why remove this check? We only need as many nodes as we have requested replicas in the `filtered` list. (That being said, I believe we can rely on the length of the `filtered` list rather than maintain a separate count.

Reply:
The idea is that getNodesForReplica() should return a full list of available nodes. And before creating replica attachments, it will decrement available attachments count. If the attachment fails(because race condition like the node has no capacity to attach more disks), try next node (here is why we need more nodes than numReplica). Let me know if this works.